openmc/deplete/operator.py

```diff
@@ -231,10 +245,18 @@ def _extract_number(self, local_mats, volume, nuclides):
             for nuc in self._burnable_nucs:
                 self.number.set_atom_density(np.s_[:], nuc, self.dilute_initial)
 
-        # Now extract the number densities and store
-        for mat in self.geometry.get_all_materials().values():
-            if str(mat.id) in local_mats:
-                self._set_number_from_mat(mat)
+        # Now extract and store the number densities
+        # From the geometry if no previous depletion results
+        if prev_res == None:
```

Comment:
An explanation for these comments:  It's considered good practice in Python to do these `None` checks with `is` rather than `==`.  It might not be super relevant for this particular case, but it's a good habit to learn.  When you define a class, the `==` operator can be evaluated however you like and you might accidentally make `== None` evaluate to True, or there might be some obscure reason you do that on purpose.  In this case, we know that the specific Python object `None` has a special meaning so it's better to use the more restrictive comparison, `is`.

Reply:
Thanks ! Learned some Python today